urla958522fc1_spirosikmd_inmdsy_08_tgz-pJ8-TesterAppJ8/src/ft2server.java

```diff
@@ -0,0 +1,65 @@
+import java.io.*;
+import java.net.*;
+public class ft2server {
+
+	public static void main(String[] args) throws IOException {
+		
+		ServerSocket ss=null;
+        try
+        {  
+            ss=new ServerSocket(8081);
+        }
+        catch(IOException e)
+        { 
+            System.out.println("couldn't listen");
+            System.exit(0);
+        }
+        Socket cs=null;
+        try
+        { 
+            cs=ss.accept();
+            System.out.println("Connection established"+cs);
+        }
+        catch(Exception e)
+        { 
+            System.out.println("Accept failed");
+            System.exit(1);
+        } 
+        PrintWriter put=new PrintWriter(cs.getOutputStream(),true);
+        
+        BufferedReader st=new BufferedReader(new InputStreamReader(cs.getInputStream()));
+        String s1=st.readLine();
+        char data[]={'"'};
+        String s = data[0]+s1+data[0];
+        System.out.println("The requested file is : "+s);
+        try
+        {
+        	FileReader fstream = new FileReader(s);
+            
+        	
+        	//DataInputStream in = new DataInputStream(fstream);
+            BufferedReader d=new BufferedReader(new FileReader(s));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *d*. The resource is referenced at lines 41, 43. The resource is closed at line 48. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *d* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).